crates/ruff/src/resolve.rs

```diff
@@ -78,7 +85,11 @@ pub fn resolve(
             "Using configuration file (via cwd) at: {}",
             pyproject.display()
         );
-        let settings = resolve_root_settings(&pyproject, Relativity::Cwd, config_arguments)?;
+        let settings = resolve_root_settings(
+            &pyproject,
+            config_arguments,
+            ConfigurationOrigin::UserSettings,
+        )?;
```

Comment:
It seems inconsistent that we don't apply the fallback behavior if a user only has the user-level configuration. It means that we won't pick up the right python version if you have:

* a user level ruff configuration
* but use a `pyproject.toml` without a `tool.ruff` section at the project level

I think it would be good to apply the `requires-python` fallback in this case as well.